Sources/StreamChat/Deprecations.swift

```diff
@@ -213,3 +213,27 @@ public typealias NotificationInviteAccepted = NotificationInviteAcceptedEvent
 
 @available(*, deprecated, renamed: "UnknownChannelEvent")
 public typealias UnknownEvent = UnknownChannelEvent
+
+public extension CurrentChatUserController {
+    @available(
+        *,
+        deprecated,
+        message: "use addDevice(_pushDevice:) instead. This deprecated function doesn't correctly support multiple push providers."
+    )
+    func addDevice(token: Data, pushProvider: PushProvider = .apn, completion: ((Error?) -> Void)? = nil) {
```

Comment:
Shall we also deprecate `PushProvider`?

Reply:
We are still using `PushProvider`, especially because `PushDevice.pushProvider` is public. I think it is still useful, even tho for now it is not exposed. But it is used internally and can be access publicly